### PR TITLE
MAT-6680: Add Read Only CQL to QDM Test Case View

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-6680](https://jira.cms.gov/browse/MAT-6680)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/RightPanel/RightPanelNavTabs.tsx [hunk 1]
```diff
@@ -17,6 +17,14 @@ export default function RightPanelNavTabs(props: NavTabProps) {
       }}
       type="D"
     >
+      <Tab
+        tabIndex={0}
+        aria-label="Measure CQL View Only tab panel"
+        type="B"
+        label={`Measure CQL (View Only)`}
```
Comment (adongare) [inline src/components/editTestCase/qdm/RightPanel/RightPanelNavTabs.tsx]:
```suggestion
        label="Measure CQL (View Only)"
```

File: src/components/editTestCase/qdm/RightPanel/RightPanel.tsx [hunk 2]
```diff
@@ -53,6 +55,23 @@ const RightPanel = ({
         {activeTab === "details" && (
           <DetailsSection canEdit={canEdit} measureName={measureName} />
         )}
+
+        {activeTab === "measurecql" &&
+          (!cqlErrors ? (
```
Comment (adongare) [inline src/components/editTestCase/qdm/RightPanel/RightPanel.tsx:60]: I think might be helpful displaying CQL all the time even there are errors

File: src/components/editTestCase/qdm/RightPanel/RightPanel.tsx [hunk 3]
```diff
@@ -53,6 +55,23 @@ const RightPanel = ({
...
+            <div data-testid="test-case-cql-editor" id="test-case-cql-editor">
+              <MadieEditor
+                value={measureCql}
+                height="52vh"
+                readOnly={true}
+                validationsEnabled={false}
+              />
```
Comment (RohitKandimalla) [inline src/components/editTestCase/qdm/RightPanel/RightPanel.tsx:72]: Can we also address the 3rd AC from the Story ?
ensuring that the view only area is horizontally scrollable

File: src/components/editTestCase/qdm/RightPanel/RightPanel.tsx (hunk 2 above)
Comment (adongare) [reply src/components/editTestCase/qdm/RightPanel/RightPanel.tsx:60]: please ignore my above suggestion. QICore is doing the same. I'm not sure why we should hide CQL tab when there are errors.

Comment (jkotanchik-SB) [reply src/components/editTestCase/qdm/RightPanel/RightPanel.tsx:60]: > please ignore my above suggestion. QICore is doing the same. I'm not sure why we should hide CQL tab when there are errors.

Me either. Best I can come up with is maybe we don't have the tooling in place to display those errors in madie-patient.

Comment (RohitKandimalla) [reply src/components/editTestCase/qdm/RightPanel/RightPanel.tsx:60]: I agree with you Ashok, I don't see an issue with displaying CQL with errors. May be we can take this back to PO's, I do see that we have a error message always on top reminding them that CQL has errors. 

image

Comment (adongare) [reply src/components/editTestCase/qdm/RightPanel/RightPanel.tsx:60]: sure, will talk to Christina

File: src/components/editTestCase/qdm/RightPanel/RightPanel.tsx (hunk 3 above)
Comment (ethankaplan) [reply src/components/editTestCase/qdm/RightPanel/RightPanel.tsx:72]: Removed after discussion

**Review by adongare — APPROVED** [review]

**Pull request merged**